src/algorithms/tracking/TrackPropagation.cc

```diff
@@ -54,26 +54,43 @@ namespace eicrecon {
 
 
 
-    std::vector<std::unique_ptr<edm4eic::TrackPoint>>
+    std::unique_ptr<edm4eic::TrackSegmentCollection>
     TrackPropagation::propagateMany(std::vector<const ActsExamples::Trajectories *> trajectories,
-                                    const std::shared_ptr<const Acts::Surface> &targetSurf) {
+                                    const std::vector<std::shared_ptr<Acts::Surface>> &target_surface_list) {
+
         // output collection
-        std::vector<std::unique_ptr<edm4eic::TrackPoint>> track_points;
+        auto propagated_tracks = std::make_unique<edm4eic::TrackSegmentCollection>();
         m_log->trace("Track propagation event process. Num of input trajectories: {}", std::size(trajectories));
 
+        // Create Acts logger for this event
+        m_acts_log = std::move(eicrecon::getSpdlogLogger(m_log, {"Propagation reached the step count limit", "Step failed with EigenStepperError"}));
+        m_log->info("Acts logger now active");
+
         // Loop over the trajectories
         for (size_t traj_index = 0; traj_index < trajectories.size(); traj_index++) {
             auto &traj = trajectories[traj_index];
             m_log->trace("  Trajectory object # {}", traj_index);
 
-            auto result = propagate(traj, targetSurf);
-            if(!result) continue;
+            auto this_propagated_track = propagated_tracks->create();
+            for(size_t isurf = 0; auto surf: target_surface_list) {
+
+                auto prop_point = propagate(traj, surf);
+                if(!prop_point) continue;
+#if EDM4EIC_VERSION_MAJOR >= 3
+//                prop_point->surface = m_target_surface_ID[isurf];
```

Comment:
The reason for no diffs is because of the disabled lines in https://github.com/eic/EICrecon/pull/1021/files#diff-8d72010a235f53e207b8ccc1bccc1579c12a373bc56bf3dbc13798aa45827564L294-L297.